src/Rules/Functions/PrintfArrayParametersRule.php

```diff
@@ -0,0 +1,143 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\IntegerRangeType;
+use function array_key_exists;
+use function count;
+use function sprintf;
+use function strtolower;
+
+/**
+ * @implements Rule<Node\Expr\FuncCall>
+ */
+class PrintfArrayParametersRule implements Rule
+{
+
+	private const FORMAT_ARGUMENT_POSITIONS = [
+		'vprintf' => 0,
+		'vsprintf' => 0,
+	];
+	private const MINIMUM_NUMBER_OF_ARGUMENTS = [
+		'vprintf' => 1,
+		'vsprintf' => 1,
+	];
+
+	public function __construct(private PrintfHelper $printfHelper)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!($node->name instanceof Node\Name)) {
+			return [];
+		}
+
+		$name = strtolower((string) $node->name);
```

Comment:
You need to fetch FunctionReflection from ReflectionProvider and ask about its name.